[CS2113-W10-2] MojoHR

FarmZH98 opened the pull request:
helps an HR personnel keep track of all her emails in the HR department. It allows her to classify her emails and reply emails. It will not give her a notification when a new email comes. 
It only allows for one user at a time, multiple users cannot log in at the same time to use the application.

yanli1215 committed:
add utilities diagram

yanli1215 committed:
Add FindCommand test

yanli1215 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp:
  Update user guide

Chihui8199 committed:
Updated the Login Component of the Developer Guide

yanli1215 committed:
add Utility component and manual testing instructions

Chihui8199 committed:
Insert diagram into Login Component developer guide

yanli1215 committed:
fix style errors

FarmZH98 committed:
Added Junit Tests and fixed minor bugs and code violations

FarmZH98 committed:
fixed bug in edit and send command

Chihui8199 committed:
Add how list command works

FarmZH98 committed:
Added JunitTest to EditCommand

Chihui8199 committed:
Update Sequence Diagram for ListCommand

jalvinchan committed:
Merge pull request #69 from Chihui8199/Developer-Guide

Updated the Login Component of the Developer Guide

jalvinchan committed:
Merge branch 'master' into master

jalvinchan committed:
Merge pull request #70 from yanli1215/master

Add Utilities component, manual test instructions in DG

jalvinchan committed:
FIx code style errors

jalvinchan committed:
Merge pull request #71 from FarmZH98/DeveloperGuide&Testing

Fixed Bugs and modified functions for integration

FarmZH98 committed:
draft email class diagram

FarmZH98 committed:
Merge branch 'master' into DeveloperGuide&Testing

jalvinchan committed:
Update storage after tag

jalvinchan committed:
Merge pull request #72 from jalvinchan/master

Update storage after tag

jalvinchan committed:
Fix checkstyle errors

jalvinchan committed:
Merge pull request #73 from jalvinchan/master

Fix checkstyle errors

jalvinchan committed:
Update developer guide with architect explanation, tag implementation.

jalvinchan committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp

FarmZH98 committed:
Updated EmailClassDiagram and NFR in DG

FarmZH98 committed:
added Implementation for ComposeCommand

jalvinchan committed:
Merge pull request #75 from FarmZH98/DeveloperGuide&Testing

Developer Guide updates

gcksean commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,82 @@
+
+The details needed are receiver email, subject and content. The time the draft was composed would be saved automatically.
 
+The sequence diagram roughly shows how the `compose` operation works.
+![ComposeCommand Sequence Diagram](Sequence Diagrams/ComposeCommand.png)
```
![image](https://user-images.githubusercontent.com/56064050/113082529-3cf74400-920d-11eb-9e9d-98eb37b1c208.png)

Is the return arrow for "7" necessary since the rest of the methods do not include return arrows?

gcksean commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,82 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+![Architect class diagram](UML diagrams/OverviewClassDiagram.png)
+![Architect sequence diagram](UML diagrams/OverviewSequenceDiagram.png)
```
![image](https://user-images.githubusercontent.com/56064050/113082860-cc9cf280-920d-11eb-9d18-9b4f67a5beb0.png)

Is the placement for the Object Deletion notation for the Command Object correct? Would it be clearer if it is placed slightly below the activation bar?

jalvinchan committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp

# Conflicts:
#	docs/DeveloperGuide.md

FarmZH98 committed:
Merge pull request #74 from jalvinchan/master

Update developer guide

chenling1022 commented:
[](url)
<img width="578" alt="Screenshot 2021-04-01 at 10 13 57 AM" src="https://user-images.githubusercontent.com/59994961/113234390-c113ff00-92d3-11eb-8b93-3b3e888c0d03.png"> 
should be userId:String ?

chenling1022 commented:
<img width="638" alt="Screenshot 2021-04-01 at 10 14 56 AM" src="https://user-images.githubusercontent.com/59994961/113234461-e0129100-92d3-11eb-85be-c09528633cd0.png">
probably should consider removing the icon

chenling1022 commented:
<img width="512" alt="Screenshot 2021-04-01 at 10 15 42 AM" src="https://user-images.githubusercontent.com/59994961/113234479-eb65bc80-92d3-11eb-8a29-4be6a07fd37e.png">
deletion of object?

chenling1022 commented:
<img width="699" alt="Screenshot 2021-04-01 at 10 18 34 AM" src="https://user-images.githubusercontent.com/59994961/113234526-033d4080-92d4-11eb-8d44-9b3f47d9f3db.png">
it seems like the return values' names are not necessary

tehtea commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,118 @@
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
+![Architect class diagram](UML diagrams/OverviewClassDiagram.png)
```
The architecture diagram seems too granular, e.g. Email and Command seem to be individual classes rather than an entire component. Also, such apps usually use n-tier architecture so it might be better to make the layered nature more explicit in the architecture diagram.

tehtea commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,118 @@
+* The Command component exposes its functionality through different command classes that inherit from a base Command class.
+
+**How the architecture components interact with each other**
+![Architect sequence diagram](Sequence Diagrams/OverviewSequenceDiagram.png)
```
![image](https://user-images.githubusercontent.com/24218271/113238806-148a4b00-92dc-11eb-9d43-10b73e09ee59.png)

The `providedLoginInfo` returned from LoginController - should it be a dotted arrow instead?

zihan9485 replied:
Should there be a colon ':' before the Mojo class?
![image](https://user-images.githubusercontent.com/77315806/113331717-dc721f00-9352-11eb-89f1-76e0905a7889.png)

tehtea commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,118 @@
+5. `TagCommand` updates the hard disk through `Storage`.
+
+The sequence diagram shows how the <code> list(type) </code> operation work.
+![Tag Sequence Diagram](Sequence Diagrams/TagSequenceDiagram.png)
```
![image](https://user-images.githubusercontent.com/24218271/113239868-279e1a80-92de-11eb-9375-e73c9433e225.png)
Perhaps each of the command calls ought to have parenthesis beside them, e.g. `execute()` instead of `execute`.
Also, each of the objects do not have a leading colon, e.g. it should be `:Actor` instead of `Actor`.

tehtea replied:
![image](https://user-images.githubusercontent.com/24218271/113240136-af842480-92de-11eb-9653-8445b50b466b.png)
Perhaps there should be dotted return arrows from `printFeedback()` and `updateAllTypeEmails()`. Even if the functions return void, it denotes the return of program control.

tehtea commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,118 @@
+The details needed are receiver email, subject and content. The time the draft was composed would be saved automatically.
+
+The sequence diagram roughly shows how the `compose` operation works.
+![ComposeCommand Sequence Diagram](Sequence Diagrams/ComposeCommand.png)
```
![image](https://user-images.githubusercontent.com/24218271/113240295-0689f980-92df-11eb-89c1-9068f5b6fac0.png)
Same for this, there should be dotted return arrows from all function calls. Also, is `ComposeCommand` deleted after `printEmailDraft()` ends?

tehtea reviewed (commented):
I like how comprehensive the documentation is, good job! Just some minor nitpicks with the diagrams but the overall quality and effort is very good IMO.

zihan9485 commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,118 @@
+### Utilities Component
+The Utilities component contains the main classes that run the main functions of Mojo.
+
+![Utility diagram](UML diagrams/Utilities.jpeg)
```
Should the class icons 'C' be removed?
![image](https://user-images.githubusercontent.com/77315806/113331154-29a1c100-9352-11eb-8cc2-a49cd13c40f7.png)

blank-bank reviewed (commented):
![image](https://user-images.githubusercontent.com/70243841/113378238-1b32c400-93a9-11eb-8fe8-05a858757dcb.png)

From the looks of the overall diagram, the system looks highly coupled. Maybe can consider an implementation that mojo does not access the storage directly. The facade pattern for architectural design can be used in this scenario.

jianningzhuang commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,118 @@
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
+![Architect class diagram](UML diagrams/OverviewClassDiagram.png)
+The Architecture Diagram above gives an overview of the different components in the application. 
```
![components](https://user-images.githubusercontent.com/75778753/113382495-29d2a880-93b4-11eb-9ae8-c859758f68c4.PNG)

Great top down structure for your architecture diagram. However, some components mentioned are not explicitly shown or grouped together in the diagram, for example, the Login and Utilities components. Perhaps you could also consider grouping the Email and EmailManager class as a single component in your diagram.

jianningzhuang replied:
The text after the diagram seems to wrap around the image as well!

jianningzhuang commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,118 @@
+
+![Utility diagram](UML diagrams/Utilities.jpeg)
+
+The Utilities Class Diagram given above shows how the classes in the Utilities component interact with each other and classes from other component.
```
![util](https://user-images.githubusercontent.com/75778753/113385085-fc88f900-93b9-11eb-8c60-e573ad1008af.PNG)

Perhaps the Icons for class diagrams should be disabled to follow standard UML? Including the red and green checkboxes.

Also, could the font size of the methods and attributes be increased for better readability?

jianningzhuang commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,118 @@
+
+### Email Component
+The email component consists of classes that are involved with all the emails
+![Email Component Class Diagram](UML diagrams/EmailClassDiagram.png)
```
![email](https://user-images.githubusercontent.com/75778753/113385407-ab2d3980-93ba-11eb-8e69-00c80780d6c2.PNG)

Similar for the Email component, perhaps the Icons for class diagrams should be disabled to follow standard UML? Including the red and green checkboxes.

Also, could the font size of the methods and attributes be increased for better readability?

jianningzhuang commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,118 @@
+
+The details needed are receiver email, subject and content. The time the draft was composed would be saved automatically.
+
+The sequence diagram roughly shows how the `compose` operation works.
```
![seq](https://user-images.githubusercontent.com/75778753/113385626-0e1ed080-93bb-11eb-9f20-5775d2a1f6fb.PNG)

Should sequence diagrams contain numberings?
Also, perhaps you can include dotted lines to show that control is returned and possibly some return value.

FarmZH98 committed:
Updated User Guide

FarmZH98 committed:
Merge pull request #77 from FarmZH98/DeveloperGuide&Testing

Updated User Guide

Chihui8199 committed:
Clean some parts

Chihui8199 committed:
no message

Chihui8199 committed:
Edits such that when user enters the wrong information it returns to the main login page

Chihui8199 committed:
Update error message

Chihui8199 committed:
Added Log off message to ExitCommand

Chihui8199 committed:
Add some assertion

Chihui8199 committed:
Update mistakes in sequence Diagrams

Chihui8199 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp into Developer-Guide

* 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp: (23 commits)
  Updated User Guide
  added Implementation for ComposeCommand
  Updated EmailClassDiagram and NFR in DG
  Update developer guide with architect explanation, tag implementation.
  Fix checkstyle errors
  Update storage after tag
  draft email class diagram
  FIx code style errors
  Added JunitTest to EditCommand
  fixed bug in edit and send command
  Added Junit Tests and fixed minor bugs and code violations
  fix style errors
  add Utility component and manual testing instructions
  Add FindCommand test
  add utilities diagram
  Update user guide
  added some logging to track errors
  added assertion
  added call method to commands that require update in storage
  minor code change
  ...

# Conflicts:
#	docs/DeveloperGuide.md

Chihui8199 committed:
Update Implementation Section of developer guide

Chihui8199 committed:
Merge branch 'Bugs' into Developer-Guide

* Bugs:
  Add some assertion
  Added Log off message to ExitCommand
  Update error message
  Edits such that when user enters the wrong information it returns to the main login page
  no message

Chihui8199 committed:
Update loginUi menu

Chihui8199 committed:
Merge branch 'Bugs'

* Bugs:
  Update loginUi menu

Chihui8199 committed:
Further updates to implementation portion (login) of developer guide

Chihui8199 committed:
Update loginUi user interface

Chihui8199 committed:
Update user guide

Chihui8199 committed:
Merge branch 'Developer-Guide'

* Developer-Guide:
  Update user guide
  Further updates to implementation portion (login) of developer guide

Chihui8199 committed:
Merge branch 'Bugs'

* Bugs:
  Update loginUi user interface

FarmZH98 committed:
Merge pull request #79 from Chihui8199/master

Update to developer guide, and fixed some bugs

AlexanderTanJunAn commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,119 @@
+The sequence diagram shows how the <code> list(type) </code> operation work.
+![Tag Sequence Diagram](Sequence Diagrams/TagSequenceDiagram.png)
+
+###ComposeCommand
```
![Screenshot 2021-04-03 at 12 30 07 PM](https://user-images.githubusercontent.com/65282395/113468369-551ecb80-9478-11eb-9bfe-e67eaf611e81.png)

Should there be a space between ### and ComposeCommand?

AlexanderTanJunAn commented on docs/DeveloperGuide.md:
`````diff
@@ -23,12 +131,274 @@
+Enter password: `5678`
+
+Expected:
+```` Hello! I'm MojoHR
`````
![Screenshot 2021-04-03 at 12 34 19 PM](https://user-images.githubusercontent.com/65282395/113468455-eb52f180-9478-11eb-8833-eea690a8dac1.png)

Should the format for code fencing be 4 "````" ? If changed to the correct 3 "```" The expected output will be much neater and standardised with the other outputs.

AlexanderTanJunAn commented on docs/DeveloperGuide.md:
```diff
@@ -23,12 +131,274 @@
+
+Enter password: `1234`
+
+Expected:
```
Same here as well, should be 3 "```".

AlexanderTanJunAn commented on docs/DeveloperGuide.md:
```diff
@@ -23,12 +131,274 @@
+Expected: `No matching emails found.`
+
+
+##Resetting the password
```
![Screenshot 2021-04-03 at 12 39 06 PM](https://user-images.githubusercontent.com/65282395/113468590-96fc4180-9479-11eb-8cef-9a5486dc95f2.png)

Same as above, should there be a space between ## and Resetting?

AlexanderTanJunAn commented on docs/DeveloperGuide.md:
```diff
@@ -23,12 +131,274 @@
+The number of wrong attempt is 3. 
+If the old password is wrong for 3 times, the program will output `Sorry your old password is wrong. Please try again!(0 times left!)` and back to main menu.
+
+###Test case 1: Typing wrong old password for 3 times
```
![Screenshot 2021-04-03 at 12 39 49 PM](https://user-images.githubusercontent.com/65282395/113468600-af6c5c00-9479-11eb-8ce6-974a341ea14b.png)

As above.

AlexanderTanJunAn commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,119 @@
+The details needed are receiver email, subject and content. The time the draft was composed would be saved automatically.
+
+The sequence diagram roughly shows how the `compose` operation works.
+![ComposeCommand Sequence Diagram](Sequence Diagrams/ComposeCommand.png)
```
![Screenshot 2021-04-03 at 12 43 12 PM](https://user-images.githubusercontent.com/65282395/113468670-30c3ee80-947a-11eb-9941-769f9091a7bc.png)

Should there be dotted return arrows from function calls to denote the return of program control?

AlexanderTanJunAn replied:
![Screenshot 2021-04-03 at 12 54 11 PM](https://user-images.githubusercontent.com/65282395/113468892-bbf1b400-947b-11eb-9e1c-b638863e02f1.png)

Perhaps there should be colons before Actor, CompaseCommand etc. to denote an unnamed instance of the class?

AlexanderTanJunAn commented on docs/DeveloperGuide.md:
```diff
@@ -1,11 +1,119 @@
+### Utilities Component
+The Utilities component contains the main classes that run the main functions of Mojo.
+
+![Utility diagram](UML diagrams/Utilities.jpeg)
```
![Screenshot 2021-04-03 at 12 48 47 PM](https://user-images.githubusercontent.com/65282395/113468790-f018a500-947a-11eb-8b57-ab04c4028cd2.png)

Should there be 2 "throws" in the class diagram? Also the font might be a bit small and hard for some people to read.

yanli1215 committed:
update introduction and quick start

AlexanderTanJunAn commented:
Overall very well documented! Just some minor format issues to make the overall DG neater and easier to read. Good job!!! :DD

yanli1215 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp:
  Update user guide
  Update loginUi user interface
  Further updates to implementation portion (login) of developer guide
  Update loginUi menu
  Update Implementation Section of developer guide
  Update mistakes in sequence Diagrams
  Add some assertion
  Added Log off message to ExitCommand
  Update error message
  Edits such that when user enters the wrong information it returns to the main login page
  no message
  Clean some parts
  Updated User Guide
  added Implementation for ComposeCommand
  Updated EmailClassDiagram and NFR in DG
  Update developer guide with architect explanation, tag implementation.
  Fix checkstyle errors
  Update storage after tag
  draft email class diagram

# Conflicts:
#	docs/UserGuide.md

yanli1215 committed:
Merge pull request #80 from yanli1215/master

update UG

Chihui8199 committed:
Update portfolio page

Chihui8199 committed:
this commit fixes issue #107 on Navigalibility

Chihui8199 committed:
This commit resolves issue #98

fix bug

Chihui8199 committed:
Allow user to enter email types in list command as capital letters

This commit fixes issue #111

yanli1215 committed:
solve json data format issue

yanli1215 committed:
add get to start in UG

yanli1215 committed:
Merge pull request #113 from Chihui8199/Bugs

Fix issues from practical dry run today

jalvinchan committed:
Merge pull request #114 from yanli1215/master

solve #100, #92, #99 issue

jalvinchan committed:
Enable assertions, fix checkstyle

jalvinchan committed:
Add read to storage.

Fix issue #109

jalvinchan committed:
Update data file to include read

jalvinchan committed:
Update storage to include read

jalvinchan committed:
Change tag command to single step

Change tag command logic.
Update user guide for single step tag
Fix #110

Update User guide to inform user to list before read/tag
Fix #102

yanli1215 committed:
Merge pull request #115 from jalvinchan/master

Fix PED issues

yanli1215 committed:
update command summary and FAQ

Chihui8199 committed:
Clean up and update the PPP

Chihui8199 committed:
This commit fixed issue #76

Updated value proposition of the developer guide

Chihui8199 committed:
This commit fixes issue #76

Included product name in developer guide tittle

Chihui8199 committed:
This commit fixes issue #76

Fixed formatting issue of Compose Command

Chihui8199 committed:
Some formatting errors

Chihui8199 committed:
Fix CheckStyle errors

Chihui8199 committed:
Change list emails to list allemails  as suggested during the practical dry run

Chihui8199 committed:
Fix the developer guide and userguide from emails to allemails

Chihui8199 committed:
Created and updated use command summary table

Chihui8199 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp into Developer-Guide

* 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp:
  Change tag command to single step
  Update storage to include read
  Update data file to include read
  Add read to storage.
  Enable assertions, fix checkstyle
  add get to start in UG
  solve json data format issue
  Allow user to enter email types in list command as capital letters
  This commit resolves issue #98
  this commit fixes issue #107 on Navigalibility

# Conflicts:
#	docs/DeveloperGuide.md
#	docs/UserGuide.md

Chihui8199 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp into Bugs

* 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp:
  Change tag command to single step
  Update storage to include read
  Update data file to include read
  Add read to storage.
  Enable assertions, fix checkstyle
  add get to start in UG
  solve json data format issue

# Conflicts:
#	src/main/java/seedu/duke/login/LoginManager.java
#	src/main/java/seedu/duke/login/LoginUi.java

Chihui8199 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp into PPP---portfolio

* 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp:
  Change tag command to single step
  Update storage to include read
  Update data file to include read
  Add read to storage.
  Enable assertions, fix checkstyle
  add get to start in UG
  solve json data format issue
  Allow user to enter email types in list command as capital letters
  This commit resolves issue #98
  this commit fixes issue #107 on Navigalibility
  update introduction and quick start
  Update user guide
  Update loginUi user interface
  Further updates to implementation portion (login) of developer guide

jalvinchan committed:
Merge pull request #116 from Chihui8199/PPP---portfolio

PPP

jalvinchan committed:
Merge pull request #117 from Chihui8199/Developer-Guide

Edits to developer and user guide

jalvinchan committed:
Merge pull request #118 from Chihui8199/Bugs

Bugs: CheckStyle errors + list emails to list allemails

yanli1215 committed:
add V2 user stories

yanli1215 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp:
  Created and updated use command summary table
  Fix the developer guide and userguide from emails to allemails
  Change list emails to list allemails  as suggested during the practical dry run
  Fix CheckStyle errors
  Some formatting errors
  This commit fixes issue #76
  This commit fixes issue #76
  This commit fixed issue #76
  Clean up and update the PPP
  Update portfolio page

FarmZH98 committed:
solve issue #108

FarmZH98 committed:
implemented check email validity based on 4 email ID (gmail, yahoo, outlook, hotmail)

yanli1215 committed:
fix merge confict

yanli1215 committed:
fix ./runtest.sh bugs

Chihui8199 committed:
Enter product name in user guide

Chihui8199 committed:
no message

Chihui8199 committed:
Merge branch 'Developer-Guide'

* Developer-Guide:
  Enter product name in user guide

jalvinchan committed:
Merge pull request #119 from yanli1215/master

user stories V2 in DG, update command summary and FAQ in UG

jalvinchan committed:
Merge pull request #120 from FarmZH98/DeveloperGuide&Testing

Check Email validity

jalvinchan committed:
Add content page, simplify diagrams

jalvinchan committed:
Merge branch 'master' into guides

yanli1215 committed:
createEmailObject lack a "tag" field

yanli1215 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp:
  implemented check email validity based on 4 email ID (gmail, yahoo, outlook, hotmail)
  solve issue #108

yanli1215 committed:
fix checkStyle bugs

jalvinchan committed:
Check recipient emails are in the storage before sending

jalvinchan committed:
Merge pull request #122 from yanli1215/master

fix bugs when add tag to local disk

jalvinchan committed:
tags are saved in storage now

jalvinchan committed:
Merge branch 'bugs' into send_check

jalvinchan committed:
Fix bugs with sending multiple emails containing duplicate index

jalvinchan committed:
updated user guide, add ppp

yanli1215 committed:
Merge pull request #121 from jalvinchan/guides

Guides

jalvinchan committed:
Fix login bug, fix checkstyle

FarmZH98 committed:
checks for validity in email address, subject and content when composing email, fires up warning if error is found (fix #84)

FarmZH98 committed:
similar to compose, show warning message for edit command

FarmZH98 committed:
Fixed bugs for SendCommand and made Send and EditCommand more error proof

FarmZH98 committed:
Merge pull request #123 from jalvinchan/send_check

check recipient email address

FarmZH98 committed:
fixed code style violations

FarmZH98 committed:
Merge branch 'master' into DeveloperGuide&Testing

# Conflicts:
#	src/main/java/seedu/duke/command/ComposeCommand.java

FarmZH98 committed:
will not send if email address is not valid.
Can still compose and edit with invalid email address but will fire up warning.
Can still send if content and subject is empty but will fire up warning.

FarmZH98 committed:
fixed minor code violation

yanli1215 committed:
Merge pull request #125 from FarmZH98/DeveloperGuide&Testing

Fixed bugs and issues

Chihui8199 committed:
1. Remove test file for outlook - no longer using domain @outlook.com
2. Change error message for wrong email account
3. Only allows users to create account from one of the 4 domains agreed upon

jalvinchan committed:
Update recipient inbox in storage after send

jalvinchan committed:
Merge pull request #127 from Chihui8199/Bugs

Email Validity Check

jalvinchan committed:
Merge branch 'master' into send_inbox

# Conflicts:
#	data/test1@outlook.com.json

jalvinchan committed:
Fix read bug, fix codestyle

Fix #124

FarmZH98 committed:
PPP

FarmZH98 committed:
Merge branch 'master' into DeveloperGuide&Testing

FarmZH98 committed:
Merge pull request #128 from jalvinchan/send_inbox

Send to recpient inbox

FarmZH98 committed:
added user stories and fixed minor formatting issues

FarmZH98 committed:
Merge branch 'master' into DeveloperGuide&Testing

FarmZH98 committed:
added documentation and fixed bug in sort time comparator

FarmZH98 committed:
Added sort and sendCommand tests

FarmZH98 committed:
fixed documentation violations

FarmZH98 committed:
fixed minor documentation violation

FarmZH98 committed:
fixed coding violations

FarmZH98 committed:
fixed coding violations

jalvinchan committed:
Remove static keyword

Guys please dont use static keywords unless it is parser
Fix issue #129 part 1

jalvinchan committed:
Fix issue #129 part 2

FarmZH98 committed:
linked AboutUs to PPP

FarmZH98 committed:
updated PPP

jalvinchan committed:
Merge pull request #130 from FarmZH98/DeveloperGuide&Testing

Developer guide & testing

FarmZH98 committed:
refactored methods to improve readability

jalvinchan committed:
Merge branch 'master' into send_inbox

# Conflicts:
#	src/test/java/seedu/duke/command/SortCommandTest.java

jalvinchan committed:
Merge pull request #131 from FarmZH98/DeveloperGuide&Testing

Updated documentation and refactored code

jalvinchan committed:
Merge branch 'master' into send_inbox

# Conflicts:
#	src/main/java/seedu/duke/command/SendCommand.java

FarmZH98 committed:
Merge pull request #132 from jalvinchan/send_inbox

Fix send command deleting sender's email storage

Chihui8199 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp: (36 commits)
  refactored methods to improve readability
  updated PPP
  linked AboutUs to PPP
  Fix issue #129 part 2
  Remove static keyword
  fixed coding violations
  fixed coding violations
  fixed minor documentation violation
  fixed documentation violations
  Added sort and sendCommand tests
  added documentation and fixed bug in sort time comparator
  added user stories and fixed minor formatting issues
  PPP
  Fix read bug, fix codestyle
  Update recipient inbox in storage after send
  1. Remove test file for outlook - no longer using domain @outlook.com 2. Change error message for wrong email account 3. Only allows users to create account from one of the 4 domains agreed upon
  fixed minor code violation
  will not send if email address is not valid. Can still compose and edit with invalid email address but will fire up warning. Can still send if content and subject is empty but will fire up warning.
  fixed code style violations
  Fixed bugs for SendCommand and made Send and EditCommand more error proof
  ...

# Conflicts:
#	docs/UserGuide.md
#	src/main/java/seedu/duke/login/LoginUi.java

FarmZH98 committed:
fix #133

FarmZH98 committed:
fix #134, and added edit and sort commands.

FarmZH98 committed:
modified help menu for multiple indices in sendCommand (fixed #137 in previous commit too)

FarmZH98 committed:
fixed code violation

FarmZH98 committed:
fixed minor bug in numberCommand

FarmZH98 committed:
minor change in help menu for sendComand

jalvinchan committed:
Merge pull request #138 from FarmZH98/DeveloperGuide&Testing

Fixing various issues

jalvinchan committed:
Fix issue #135, #139. Add tag command test. Add assertions

jalvinchan committed:
Merge branch 'master' into send_inbox

jalvinchan committed:
Fix checkstyle error

FarmZH98 committed:
updated user guide and fix bugs in compose

FarmZH98 committed:
fix code violation

FarmZH98 committed:
fix code violation

yanli1215 committed:
Merge pull request #143 from jalvinchan/send_inbox

Fix tag bugs, error message

yanli1215 committed:
Check the validation of the password

yanli1215 committed:
Add comment

yanli1215 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp: (36 commits)
  Fix checkstyle error
  Fix issue #135, #139. Add tag command test. Add assertions
  minor change in help menu for sendComand
  fixed minor bug in numberCommand
  fixed code violation
  modified help menu for multiple indices in sendCommand (fixed #137 in previous commit too)
  fix #134, and added edit and sort commands.
  fix #133
  refactored methods to improve readability
  updated PPP
  linked AboutUs to PPP
  Fix issue #129 part 2
  Remove static keyword
  fixed coding violations
  fixed coding violations
  fixed minor documentation violation
  fixed documentation violations
  Added sort and sendCommand tests
  added documentation and fixed bug in sort time comparator
  added user stories and fixed minor formatting issues
  ...

# Conflicts:
#	src/main/java/seedu/duke/utilities/Parser.java
#	src/main/java/seedu/duke/utilities/Ui.java

yanli1215 committed:
fix style errors

FarmZH98 committed:
fix further code violations and user guide

FarmZH98 committed:
Merge branch 'master' into DeveloperGuide&Testing

# Conflicts:
#	src/main/java/seedu/duke/utilities/Ui.java

FarmZH98 committed:
fix readCommandTest

Chihui8199 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp:
  Fix checkstyle error
  Fix issue #135, #139. Add tag command test. Add assertions
  minor change in help menu for sendComand
  fixed minor bug in numberCommand
  fixed code violation
  modified help menu for multiple indices in sendCommand (fixed #137 in previous commit too)
  fix #134, and added edit and sort commands.
  fix #133

Chihui8199 committed:
Added new line at end of file
This commit fixes #150

yanli1215 committed:
Merge pull request #144 from FarmZH98/DeveloperGuide&Testing

updated user guide and fix bugs in compose

yanli1215 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp:
  fix readCommandTest
  fix further code violations and user guide
  fix code violation
  fix code violation
  updated user guide and fix bugs in compose

FarmZH98 committed:
Merge pull request #145 from yanli1215/master

check the password's validation

FarmZH98 committed:
Merge pull request #151 from Chihui8199/master

New Line at end of file

FarmZH98 committed:
update DG and fixed PPP name

FarmZH98 committed:
Merge pull request #152 from FarmZH98/DeveloperGuide&Testing

update DG

FarmZH98 committed:
updated page break for PPP

FarmZH98 committed:
updated UG

Chihui8199 committed:
This commit fixes #146

yanli1215 committed:
Merge pull request #153 from Chihui8199/master

This commit fixes #146

jalvinchan committed:
update link

yanli1215 committed:
update DG and UG

jalvinchan committed:
Merge pull request #155 from jalvinchan/master

update link

yanli1215 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp:
  update link

yanli1215 committed:
update menu in DG

FarmZH98 committed:
fix #126

FarmZH98 committed:
Merge pull request #156 from yanli1215/master

update DG and UG

jalvinchan committed:
fix ppp link

jalvinchan committed:
Include code to load data files from jar

if data folder no exist, load from jar resource.
refactor storage code because i need the storage object to load the resource first before knowing the login details

jalvinchan committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp

jalvinchan committed:
Fix checkstyle, update DG

FarmZH98 committed:
Updated DG for manual testing, UG for FAQ, and composeCommand sequence diagram

FarmZH98 committed:
reformat code

FarmZH98 committed:
Merge branch 'master' into DeveloperGuide&Testing

# Conflicts:
#	docs/DeveloperGuide.md
#	docs/UserGuide.md

FarmZH98 committed:
added glossary

FarmZH98 committed:
UG FAQ supported email domains

yanli1215 committed:
Merge pull request #158 from FarmZH98/DeveloperGuide&Testing

Developer & user guide

yanli1215 committed:
add yanli.md

yanli1215 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp:
  UG FAQ supported email domains
  added glossary
  reformat code
  Updated DG for manual testing, UG for FAQ, and composeCommand sequence diagram
  fix #126
  updated UG
  updated page break for PPP

yanli1215 committed:
update dg and ug

FarmZH98 committed:
modified sequence diagram for compose command

FarmZH98 committed:
Merge pull request #159 from yanli1215/master

update UD and DG

FarmZH98 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp into DeveloperGuide&Testing

FarmZH98 committed:
update aboutUs

FarmZH98 committed:
updated UG examples and DG

Chihui8199 committed:
Updated Javadocs for Login features

yanli1215 committed:
Merge pull request #160 from FarmZH98/DeveloperGuide&Testing

update UG and DG

Chihui8199 committed:
Added javadoc for the helpCommand, ListCommand and fixed checkstyle errors

Chihui8199 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp:
  updated UG examples and DG
  update aboutUs
  modified sequence diagram for compose command
  update dg and ug
  add yanli.md
  UG FAQ supported email domains
  added glossary
  reformat code
  Updated DG for manual testing, UG for FAQ, and composeCommand sequence diagram
  fix #126
  update menu in DG
  update DG and UG
  update link
  updated UG
  updated page break for PPP

jalvinchan committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp

# Conflicts:
#	docs/AboutUs.md
#	docs/DeveloperGuide.md

jalvinchan committed:
Fix storage exception for windows

catch null pointer exception too.

Chihui8199 committed:
Update userguide with the constrains of creating new account

Chihui8199 committed:
Updated password constrains

yanli1215 committed:
update UG

jalvinchan committed:
Update UG read and tag examples

Chihui8199 committed:
Try to fix the styling issues

jalvinchan committed:
Fix read command test

Chihui8199 committed:
Merge pull request #157 from jalvinchan/master

Load resource from jar file on first usage

Chihui8199 committed:
Merge pull request #162 from yanli1215/master

update UG

Chihui8199 committed:
Merge pull request #161 from Chihui8199/master

JavaDoc for List and Help Command and Checkstyle errors

yanli1215 committed:
update UG

yanli1215 committed:
Merge pull request #163 from yanli1215/master

update UG

yanli1215 committed:
update ppp

yanli1215 committed:
Merge pull request #164 from yanli1215/master

update ppp

FarmZH98 committed:
add photos

yanli1215 committed:
update aboutme

jalvinchan committed:
update ppp

jalvinchan committed:
update about us

jalvinchan committed:
Merge pull request #167 from jalvinchan/master

update ppp and abt us

yanli1215 committed:
Merge branch 'master' into master

yanli1215 committed:
Merge pull request #166 from yanli1215/master

update aboutme

jalvinchan committed:
Try to fix windows fail test

FarmZH98 committed:
Merge branch 'master' into DeveloperGuide&Testing

# Conflicts:
#	docs/AboutUs.md

FarmZH98 committed:
update AboutUs.md

yanli1215 committed:
update ppp

FarmZH98 committed:
updated AboutUs.md

yanli1215 committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp

* 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp:
  update about us
  update ppp

yanli1215 committed:
Merge pull request #169 from yanli1215/master

update ppp

jalvinchan committed:
try fix windows error again. move data files into resources files

FarmZH98 committed:
Merge pull request #165 from FarmZH98/DeveloperGuide&Testing

add photos

jalvinchan committed:
Try to fix windows error 3

jalvinchan committed:
Fix error 4

FarmZH98 committed:
update DG

FarmZH98 committed:
Merge pull request #168 from jalvinchan/master

Try to fix windows fail test

FarmZH98 committed:
Merge branch 'master' into DeveloperGuide&Testing

FarmZH98 committed:
updated UG and sequence diagram

Chihui8199 committed:
Merge pull request #171 from FarmZH98/DeveloperGuide&Testing

updated DG

FarmZH98 committed:
markdown errors

FarmZH98 committed:
Merge pull request #172 from FarmZH98/DeveloperGuide&Testing

fixed errors

jalvinchan committed:
fix developer guide

jalvinchan committed:
Merge branch 'master' of https://github.com/AY2021S2-CS2113-W10-2/tp

# Conflicts:
#	data/12312@gmail.com.json
#	docs/DeveloperGuide.md

jalvinchan committed:
Merge pull request #175 from jalvinchan/master

fix dg

jalvinchan committed:
fix spaces

jalvinchan committed:
Merge pull request #176 from jalvinchan/master

fix spaces
